packages/components/data-resource/s3-endpoint-type.tsx

```diff
@@ -108,66 +191,141 @@ export const S3EndPointType: React.FC<S3EndpointTypeProps> = (props) => {
+                }
+                filterResource={
+                  isClusterAdmin
+                    ? undefined
+                    : (resources) => isValidNS(resources?.metadata?.name)
+                }
               />
-              <Button variant="plain" onClick={switchToSecret}>
-                {t('Switch to Secret')}
-              </Button>
             </InputGroup>
           </FormGroup>
-          <FormGroup
-            className="nb-endpoints-form-entry"
-            label={credentialField2Label}
-            fieldId="secret-key"
-          >
-            <TextInput
-              value={state.secretKey}
-              id="secret-key"
-              data-test={`${type.toLowerCase()}-secret-key`}
-              onChange={(e) => {
-                dispatch({ type: createFormAction.SET_SECRET_KEY, value: e });
-              }}
-              aria-label={t('Secret Key Field')}
-              type="password"
-            />
-          </FormGroup>
+          {showSecret && !hasReadAccess && (
+            <Alert isInline variant="danger" title={t('Restricted Access')}>
+              {t(
+                "You don't have access to read secrets from {{namespace}} namespace. Please select a different namespace.",
+                { namespace: state.secretNamespace }
+              )}
+            </Alert>
+          )}
+          {!showSecret && !hasCreateAccess && (
+            <Alert isInline variant="danger" title={t('Restricted Access')}>
+              {t(
+                "You don't have access to write secrets to {{namespace}} namespace. Please select a different namespace.",
+                { namespace: state.secretNamespace }
+              )}
+            </Alert>
+          )}
+          {showSecret ? (
+            <FormGroup
+              label={t('Secret')}
+              fieldId="secret-dropdown"
+              className="nb-endpoints-form-entry nb-endpoints-form-entry--full-width"
+              isRequired
+            >
+              <InputGroup>
+                {hasReadAccess ? (
+                  <ResourceDropdown
+                    className="nb-endpoints-form-entry__dropdown nb-endpoints-form-entry__dropdown--full-width"
+                    resource={getUpdatedSecretResource(state.secretNamespace)}
+                    resourceModel={SecretModel}
+                    onSelect={getSecrets}
+                    secondaryTextGenerator={getNamespace}
+                    initialSelection={(resources: SecretKind[]) =>
+                      resources.find(
+                        (res) => res?.metadata?.name === state.secretName
+                      )
+                    }
+                  />
+                ) : (
+                  <Dropdown
+                    className="nb-endpoints-form-entry__dropdown nb-endpoints-form-entry__dropdown--full-width mcgms-disabledDropdown__container"
+                    toggle={<DropdownToggle isDisabled />}
+                  />
+                )}
```

Comment:
```suggestion
```
add a `isDisabled` field in ResourceDropdown which will be passed as a prop...

Reply:
as u can see these are required fields for resource drop down 

![image](https://user-images.githubusercontent.com/22158580/177580629-88ccdd3e-8032-4909-97ca-e2f42de521d2.png)

I don't see any point in adding an extra parameter and then sending it when we only need a dropdown that is disabled.

we have to write 2 dropdown tags (one with disabled and one without), IMO the second one which should be disabled is better without resource information and all. simply a bare metal dropdown that is disabled.

Reply:
move this part inside ResourceDropdown and first condition will be:
if(isDisabled) return (
                  <Dropdown
                    className="nb-endpoints-form-entry__dropdown nb-endpoints-form-entry__dropdown--full-width mcgms-disabledDropdown__container"
                    toggle={<DropdownToggle isDisabled />}
                  />
                )